contracts/interchain-token-service/src/contract/execute/interceptors.rs

```diff
@@ -186,6 +192,35 @@ fn try_load_token_instance(
         .ok_or(report!(Error::TokenNotDeployed { token_id, chain }))
 }
 
+fn try_load_token_instance_or_default(
+    storage: &dyn Storage,
+    chain: &ChainSpecifier,
+    token_id: TokenId,
+) -> Result<TokenInstance, Error> {
+    let res = state::may_load_token_instance(storage, chain.name.clone(), token_id)
```

Comment:
Querying here everytime for now will be much cleaner, the additional gas cost isn't a significant concern for now

Reply:
we can't only query here, we also need to query when doing balance tracking, which shouldn't use this method, since it would then modify the balance for the origin chain